calico/felix/fetcd.py

```diff
@@ -334,6 +335,56 @@ def __init__(self, config, etcd_api, status_reporter, hosts_ipset):
         self.last_rate_log_time = monotonic_time()
         # Register for events when values change.
         self._register_paths()
+        self._usage_count_greenlet = gevent.Greenlet(
+            self._periodically_usage_count
+        )
+
+    def estimated_host_count(self):
+        """
+        Single operations to python's built-in datastructures are thread-safe.
+        You can update a map, or add something to a set or update an integer,
+        and it's atomic.
+
+        :return An aproximation to our Felix cluster count
+        """
+        return len(self.ipv4_by_hostname)
+
+    @logging_exceptions
+    def _periodically_usage_count(self):
+        """
+        Greenlet: periodically report the cluster existence to
+        projectcalico.org.
+
+        :return: Does not return, unless UsageCount disabled.
+        """
+        interval = 86400 # Once every 24 hours
+        if not self._config.USAGE_COUNT:
+            _log.info("usage count is disabled")
+            return
+
+        _log.info("Started usage count thread.  Usage count interval: %s", interval)
+
+        while True:
+            try:
+                config = self._config
+                calico_version = str(pkg_resources.require("calico")[0].version)
+                felix_count = self.estimated_host_count()
+                cluster_version = "NA"
+                usage_count(calico_version, config.HOSTNAME, config.CLUSTER_GUID, felix_count, cluster_version)
+            except Exception:
+                # except Exception: so that we don't catch SystemExit an KeyboardInterrupt. Those should kill the app.
+
+                # Jitter by 10% of interval.
+                jitter = random.random() * 0.1 * interval
+                sleep_time = interval + jitter
```

Comment:
I think the result of this jitter calculation is that it will be >24 hours (except for the case where random.random() returns 0, which is presumably pretty rare).  That means it will only be reported every other day, and we want it to be generated at least once per day.  `sleep_time = interval - jitter` would probably work.

2.4 hours might be a significantly longer jitter than required too.  Perhaps 0.05 rather than 0.1?